app-modules/campaign/src/Filament/Resources/CampaignResource/Pages/ListCampaigns.php

```diff
@@ -76,8 +78,17 @@ public function table(Table $table): Table
                     ->hidden(fn (Campaign $record) => $record->hasBeenExecuted() === true),
                 DeleteAction::make()
                     ->hidden(fn (Campaign $record) => $record->hasBeenExecuted() === true),
-            ])
-            ->bulkActions([
+            ])->filters([
+                Filter::make('My Campaigns')
+                    ->query(fn (Builder $query) => $query->where('user_id', auth()->user()->getKey())),
+                Filter::make('Enabled')
+                    ->query(fn (Builder $query) => $query->where('enabled', true)),
+                Filter::make('Completed')
+                    ->query(function (Builder $query) {
+                        $query->whereHas('actions', function (Builder $query) {
+                            $query->whereNotNull('successfully_executed_at');
+                        });
```

Comment:
I think this needs to be adjusted. I think it needs to be where doesn't have actions where `successfully_executed_at` is null.

I could be wrong but I think they way it is now it would pass if ANY of the actions are completed, not all.

Reply:
You mean like this, 
`Filter::make('Completed')
                    ->query(function (Builder $query) {
                        $query->whereDoesntHave('actions', function (Builder $query) {
                            $query->whereNull('successfully_executed_at');
                        });
                    }),`
it is not working as expected

Reply:
Hmm, if that does not work then you will need to explore other queries. Or if you believe your original query works then we can stick with it.

We should prove it out with a test for this filter though, so go ahead and create one. That will better help you craft the query as you can use the test case to confirm whether or not it is right.

The test should test that the filter causes records that you expect to be shown are shown and vice versa.

Reply:
Okay. let me create a test and will check that.

Reply:
That recommendation did end up working.